### ISSUE TITLE
Opening SFTP connection causing issues

### ISSUE THREAD

**Issue by codyaray** [StackStorm/st2contrib#505]
Discussed on Slack: https://stackstorm-community.slack.com/archives/community/p1465570980000803

tl;dr - I'm connecting from StackStorm to a Windows machine running cygwin sshd using `core.remote`. When accessing network resources with the correct permissions (username/password), we're getting an Access Denied through StackStorm but are able to access through Paramiko directly.

The test command is simply `dir \\mywindowshost\c$`. [This gist](https://gist.github.com/codyaray/9138c7a55c4fe507b98305518268324c) contains all the stuff from my investigation:

1. **st2_windows_execution.log**: debug logs from st2actionrunners
2. **paramiko_raw.py**: successful pure paramiko connection
3. **paramiko_raw_channel.py**: successful pure paramiko conn with low-level channel API like st2
4. **paramiko_st2.py**: failed wrapper around st2's ParamikoSSHClient directly showing "Access Denied"
5. **paramiko_raw_ftp_fail.py**: failed pure paramiko connection which opens ftp client like st2

Basically, commenting out `client.open_sftp()` on [this line](https://github.com/StackStorm/st2/blob/58fbabf116da8292cd6a404099072c6ca54cfc3c/st2actions/st2actions/runners/ssh/paramiko_ssh.py#L145) allows us to successfully access the network resources in this manner.

Interestingly, the Paramiko client has to be reconnected once an FTP session has been opened. Even opening and then closing the FTP client prior to executing a command causes it to fail.

@Kami suggests that we should conditionally [only open the FTP client when we need it](https://stackstorm-community.slack.com/archives/community/p1465579563000892).

**Comment by Kami** [comment]
Thanks for the report.

I will look into it and make the necessary changes next week.

**Comment by Kami** [comment]
Just noticed this was reported in st2contrib in st2 - it's a st2 runner issue.

In any case, not a big deal :)

**Comment by codyaray** [comment]
Whoops, must've had this repo open already. My bad.

Anyway, I'll update and test today.

**Comment by Kami** [comment]
No problem and keep me posted :)

**Comment by codyaray** [comment]
Ah, right, we can't test this until tomorrow since this will go into tonight's build. I just updated to the latest (1.5dev-94) and it doesn't have these changes yet. Will test tomorrow.

**Comment by Kami** [comment]
No problem - I actually need to merge change into master before packages will be built.

I will add some regression tests tomorrow and then go ahead and merge it into master.

**Comment by codyaray** [comment]
Just wanted to say this looks good. Thanks @Kami

**Comment by Kami** [comment]
@codyaray Great, thanks for confirming :)
